### PR TITLE
add Allison Numpad

### PR DESCRIPTION
**Pull request by yiancar**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/allison_numpad/info.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "allison_numpad",
+    "url": "www.yiancar-designs.com",
+    "maintainer": "Yiancar",
+    "width": 4, 
+    "height": 6.25, 
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/allison_numpad/info.json]:
```suggestion
        "LAYOUT_ortho_6x4": {
```

File: keyboards/allison_numpad/allison_numpad.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_all( \
```
Comment (zvecr) [inline keyboards/allison_numpad/allison_numpad.h]:
```suggestion
#define LAYOUT_ortho_6x4( \
```

File: keyboards/allison_numpad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all( /* Base */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/default/keymap.c]:
```suggestion
    [0] = LAYOUT_ortho_6x4( /* Base */
```

File: keyboards/allison_numpad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    KC_PSCR, KC_SLCK, KC_PAUS, TG(1),
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+    KC_P0,   KC_P0,   KC_PDOT, KC_PENT),
+
+    [1] = LAYOUT_all( /* FN1 */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/default/keymap.c]:
```suggestion
    [1] = LAYOUT_ortho_6x4( /* FN1 */
```

File: keyboards/allison_numpad/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    BL_TOGG, BL_INC,  BL_DEC,  BL_BRTG,
+    KC_TRNS, KC_UP,   KC_TRNS, KC_TRNS,
+    KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [2] = LAYOUT_all( /* Empty for dynamic keymaps */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/default/keymap.c]:
```suggestion
    [2] = LAYOUT_ortho_6x4( /* Empty for dynamic keymaps */
```

Or maybe this empty layer can be deleted?

File: keyboards/allison_numpad/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [3] = LAYOUT_all( /* Empty for dynamic keymaps */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/default/keymap.c]:
```suggestion
    [3] = LAYOUT_ortho_6x4( /* Empty for dynamic keymaps */
```

Or maybe this empty layer can be deleted?

File: keyboards/allison_numpad/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all( /* Base */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/via/keymap.c]:
```suggestion
    [0] = LAYOUT_ortho_6x4( /* Base */
```

File: keyboards/allison_numpad/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    KC_PSCR, KC_SLCK, KC_PAUS, TG(1),
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+    KC_P0,   KC_P0,   KC_PDOT, KC_PENT),
+
+    [1] = LAYOUT_all( /* FN1 */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/via/keymap.c]:
```suggestion
    [1] = LAYOUT_ortho_6x4( /* FN1 */
```

File: keyboards/allison_numpad/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    BL_TOGG, BL_INC,  BL_DEC,  BL_BRTG,
+    KC_TRNS, KC_UP,   KC_TRNS, KC_TRNS,
+    KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [2] = LAYOUT_all( /* Empty for dynamic keymaps */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/via/keymap.c]:
```suggestion
    [2] = LAYOUT_ortho_6x4( /* Empty for dynamic keymaps */
```

File: keyboards/allison_numpad/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [3] = LAYOUT_all( /* Empty for dynamic keymaps */
```
Comment (zvecr) [inline keyboards/allison_numpad/keymaps/via/keymap.c]:
```suggestion
    [3] = LAYOUT_ortho_6x4( /* Empty for dynamic keymaps */
```

File: keyboards/allison_numpad/rules.mk [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (zvecr) [inline keyboards/allison_numpad/rules.mk]:
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs

LAYOUTS = ortho_6x4
```

Comment (fauxpark) [reply keyboards/allison_numpad/rules.mk]: Judging from the image in the readme, it looks like this board can also do `numpad_6x4`. Would need to be added to allison_numpad.h and info.json.

Comment (yiancar) [reply keyboards/allison_numpad/rules.mk]: good shout

**Comment by yiancar** [comment]
This should do it!

**Pull request merged**